src/backend/distributed/metadata/metadata_sync.c

```diff
@@ -209,7 +204,16 @@ StartMetadataSyncToNode(const char *nodeNameString, int32 nodePort)
 	}
 
 	UseCoordinatedTransaction();
-	MarkNodeHasMetadata(nodeNameString, nodePort, true);
+
+	/*
+	 * We first set metadatasynced, and then hasmetadata; since setting columns for
+	 * nodes with metadatasynced==false could cause errors.
+	 * (See ErrorIfAnyMetadataNodeOutOfSync)
```

Comment:
. But as we are in a coordinated transaction the order doesn't matter so we do like this for convenience.